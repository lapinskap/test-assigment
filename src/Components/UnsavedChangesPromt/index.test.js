/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable react/prop-types */
import React from 'react';
import renderer from 'react-test-renderer';
import UnsavedChangesPrompt from './index';
import '../../tests/setupTests';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    location: {
      hash: 'test',
    },
  }),
  Prompt: ({ message }) => <div>{message}</div>,
  Link: ({ to, children, ...props }) => <a href={to} {...props}>{children}</a>,
}));

describe('<UnsavedChangesPrompt />', () => {
  const setState = jest.fn();
  const useStateSpy = jest.spyOn(React, 'useState');
  useStateSpy.mockImplementation((init) => [init, setState]);

  let message = 'Niezapisane zmiany zostanÄ… utracone.';
  let children;

  beforeEach(() => {
    children = <div />;
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly with props', () => {
    const tree = renderer
      .create(
        <UnsavedChangesPrompt
          message={message}
        >
          {children}
        </UnsavedChangesPrompt>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with null message', () => {
    message = null;
    const tree = renderer
      .create(
        <UnsavedChangesPrompt
          message={message}
        >
          {children}
        </UnsavedChangesPrompt>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
