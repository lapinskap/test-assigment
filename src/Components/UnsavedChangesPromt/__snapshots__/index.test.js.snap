// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UnsavedChangesPrompt /> renders correctly with null message 1`] = `
Array [
  <div />,
  <div
    setIsEdited={[Function]}
  />,
]
`;

exports[`<UnsavedChangesPrompt /> renders correctly with props 1`] = `
Array [
  <div>
    Niezapisane zmiany zostanÄ… utracone.
  </div>,
  <div
    setIsEdited={[Function]}
  />,
]
`;
